<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1603</width>
    <height>858</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Metalic Weathering Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="1">
     <widget class="QGroupBox" name="s_optionGB">
      <property name="title">
       <string>Weathering Models</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QTabWidget" name="s_modelsTab">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>Random Deposition</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QCheckBox" name="s_RDcheckBox">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="text">
              <string> Enable Random Deposition Model</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Number of Particles to hit mesh every update</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QSpinBox" name="s_RDNumParticles">
             <property name="maximum">
              <number>2000</number>
             </property>
             <property name="value">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0">
            <widget class="Line" name="line_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_2">
          <attribute name="title">
           <string>Random Deposition Variation</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Number of Particles to hit mesh every update</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_2">
             <property name="text">
              <string>Enable Random Deposition Variation Model</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QSpinBox" name="spinBox">
             <property name="maximum">
              <number>2000</number>
             </property>
             <property name="value">
              <number>500</number>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <spacer name="verticalSpacer_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0">
            <widget class="Line" name="line_7">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_3">
          <attribute name="title">
           <string>Ballistic Depostion</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="3" column="0">
            <widget class="QSpinBox" name="spinBox_2">
             <property name="maximum">
              <number>2000</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_3">
             <property name="text">
              <string>Enable Ballistic Deposition Model</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Number of Particles to hit mesh every update</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0">
            <widget class="Line" name="line_8">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_4">
          <attribute name="title">
           <string>Directed Percolation Depinning</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="10" column="0">
            <widget class="QRadioButton" name="s_DPDfromTexSelectRB">
             <property name="text">
              <string>From Texture</string>
             </property>
            </widget>
           </item>
           <item row="19" column="0">
            <widget class="Line" name="line_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_4">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="text">
              <string>Enable Directed Percolation Depinning Model</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="22" column="0">
            <widget class="QLabel" name="s_numParticlesLbl">
             <property name="text">
              <string>Number of particles to hit mesh every update</string>
             </property>
            </widget>
           </item>
           <item row="23" column="0">
            <widget class="QSpinBox" name="spinBox_3">
             <property name="maximum">
              <number>2000</number>
             </property>
             <property name="value">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QCheckBox" name="s_probFromPixCB">
             <property name="text">
              <string>Calculate blocked pixel probability from lattice texture</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="Line" name="line_9">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QRadioButton" name="s_DPDrandomSelectRB">
             <property name="text">
              <string>Generated Noise</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="11" column="0">
            <widget class="QPushButton" name="s_loadTexBtn">
             <property name="text">
              <string>Load Texture</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="s_texSelectLbl">
             <property name="text">
              <string>Select which texture to use</string>
             </property>
            </widget>
           </item>
           <item row="13" column="0">
            <widget class="Line" name="line_10">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="14" column="0">
            <widget class="QGroupBox" name="s_DPDSampleTypeGB">
             <property name="title">
              <string>Directed Percolation Depinning Sampling Type</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_9">
              <item row="0" column="0">
               <widget class="QRadioButton" name="s_DPDRandomRB">
                <property name="text">
                 <string>Random sampling</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QSpinBox" name="s_DPDNumSeedsSB">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>20</number>
                </property>
                <property name="singleStep">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>5</number>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QRadioButton" name="s_DPDSeedRB">
                <property name="text">
                 <string>From seed</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="s_DPDgenSeedsBtn">
                <property name="text">
                 <string>Generate New Seeds</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QGroupBox" name="s_noiseTypeGB">
             <property name="title">
              <string>Noise Type</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_8">
              <item row="5" column="0">
               <widget class="QLabel" name="s_perlinOffsetLbl">
                <property name="text">
                 <string>Offset:</string>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="s_perlinOctavesLbl">
                <property name="text">
                 <string>Octaves:</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QDoubleSpinBox" name="s_perlinSizeSB">
                <property name="value">
                 <double>16.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QRadioButton" name="s_randomNoiseRB">
                <property name="text">
                 <string>Random Noise</string>
                </property>
               </widget>
              </item>
              <item row="6" column="1">
               <widget class="QSpinBox" name="s_perlinOctavesSB">
                <property name="value">
                 <number>8</number>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QRadioButton" name="s_PerlinNoiseRB">
                <property name="text">
                 <string>Perlin Noise</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="s_randNoiseDenseLbl">
                <property name="text">
                 <string>Noise Density:</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QDoubleSpinBox" name="s_perlinOffsetSB"/>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="s_perlinSizeLbl">
                <property name="text">
                 <string>Size:</string>
                </property>
               </widget>
              </item>
              <item row="8" column="1">
               <widget class="QPushButton" name="s_genNoiseBtn">
                <property name="text">
                 <string>Generate Noise</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="s_randNoiseDenseSB">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <double>1.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QSpinBox" name="s_userDefProbSB">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>30</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="s_userDefProbLBl">
             <property name="text">
              <string>User Defined blocked pixel probability</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" rowspan="3">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="2">
     <widget class="QGroupBox" name="s_simOptionGB">
      <property name="title">
       <string>Simulation Options</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="s_modelGB">
         <property name="title">
          <string>Model</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="0">
           <widget class="QPushButton" name="s_loadMeshBtn">
            <property name="text">
             <string>Load Mesh</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="s_loadBseTexture">
            <property name="text">
             <string>Load Base Texture</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="s_rustGB">
         <property name="title">
          <string>Rust</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="1" column="2">
           <widget class="QLabel" name="s_startBlueLbl">
            <property name="text">
             <string>Blue</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QSpinBox" name="s_startRedSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>204</number>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QLabel" name="s_rustEndColourLbl">
            <property name="text">
             <string>Rust End Colour</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="s_rustStartColourLbl">
            <property name="text">
             <string>Rust Start Colour</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="s_startGreenSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>173</number>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QSpinBox" name="s_startBlueSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>54</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="s_startRedLbl">
            <property name="text">
             <string>Red</string>
            </property>
           </widget>
          </item>
          <item row="2" column="5">
           <widget class="QSpinBox" name="s_endGreenSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>69</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="s_startGreenLbl">
            <property name="text">
             <string>Green</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QSpinBox" name="s_endRedSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>139</number>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QLabel" name="s_endRedLbl">
            <property name="text">
             <string>Red</string>
            </property>
           </widget>
          </item>
          <item row="2" column="6">
           <widget class="QSpinBox" name="s_endBlueSB">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>19</number>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QLabel" name="s_endGreenLbl">
            <property name="text">
             <string>Green</string>
            </property>
           </widget>
          </item>
          <item row="1" column="6">
           <widget class="QLabel" name="s_endBlueLbl">
            <property name="text">
             <string>Blue</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="Line" name="line_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="Line" name="line_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="s_playGB">
         <property name="title">
          <string>Draw</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="2" column="0">
           <widget class="QPushButton" name="s_playToggleBtn">
            <property name="text">
             <string>Play/Pause</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="s_resetBtn">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="checkBox">
            <property name="text">
             <string>Wireframe</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="s_toggleDrawLatticeCB">
            <property name="text">
             <string>Draw DPD Lattice Texture</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>File Name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="s_saveTexName">
         <property name="inputMask">
          <string/>
         </property>
         <property name="text">
          <string>RustTexture</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="s_saveRustTextureBtn">
         <property name="text">
          <string>Save Rust Texture</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1603</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
